bundle/edu.gemini.p2checker/src/main/java/edu/gemini/p2checker/rules/general/GeneralRule.java

```diff
@@ -588,6 +587,32 @@ private boolean isAcceptable(final SPSiteQuality expected, final SPSiteQuality a
 
     };
 
+    /**
+     * Rule to indicate AGS is off.
+     */
+    private static IRule AGS_OFF_RULE = new IRule() {
+
+        private static final String AGS_OFF_MSG = "Auto guide star search disabled.";
+
+        public IP2Problems check(final ObservationElements elements)  {
+            if (elements.getTargetObsComp().isEmpty()) return null; // can't perform this check without a target environment
+
+            final P2Problems problems = new P2Problems();
+
+            if (!hasScienceObserves(elements.getSequence())) return null; //if there are not observes, ignore this check (SCT-260)
+
+            final Boolean isOff = elements.getObsContext().flatMap(
+                    ctxt -> AgsRegistrar.currentStrategyForJava(ctxt)
+            ).map(st -> st.key().id().equals("Off")).getOrElse(false);
```

Comment:
Couldn't this be done with `exists` instead of `map` and `gerOrElse(false)`?

Final line:
`.exists(_.key().id().equals("Off"));`

Reply:
Oooops... Java code, so you would need the `st -> st.....`

Reply:
Why use a string "Off" as the key? Could we at least replace it with a constant?

Reply:
Sure... something like `st -> st.key().id().equals(AgsStrategyKey.OffKey.id)`?
Since `OffKey` is a `case object`, it will be more fiddly than that, I am sure.